### PR TITLE
Websockets and v1.14

### PR DESCRIPTION
**Pull request by mendozatudares**
This continues to refactor/simplify Multicraft, adds the websocket server for WIP Javascript clients to connect to, and bumps the Bukkit API version to 1.14.

This version of the plugin is no longer compatible with the current Python client due to the new API, thus the bumped version to Multicraft 2.0.

### CODE REVIEW

File: src/main/java/com/multicraft/MultiCraftCommandExecutor.java [hunk 1]
```diff
@@ -228,98 +168,72 @@ public boolean onCommand(CommandSender sender, Command cmd, String label, String
...
+				return p.performCommand("clone " + copyArgs + pasteArgs + "masked");
 			}
 			default:
 				break;
 		}
 		p.sendMessage("No MultiCraft commands executed.");
 		return false;
 	}
+
+	private static boolean playerRequirementNotMet(Player p, Function<Player, Boolean> requirement) {
```
Comment (mitchellzhen) [inline src/main/java/com/multicraft/MultiCraftCommandExecutor.java:231]: I'm not sure if this extra wrapper function is necessary, or I think it should say which requirement was not met.

File: src/main/resources/plugin.yml [hunk 2]
```diff
@@ -27,18 +27,15 @@ commands:
...
     description: (MultiCraft) selects first corner of region
     usage: /<command>
   rloc2:
     description: (MultiCraft) selects second corner of region
     usage: /<command>
-  rrbuild:
+  rbuild:
     description: (MultiCraft) builds a structure from rloc1 to rloc2
-    usage: /<command> [material]
+    usage: /<command>
```
Comment (mitchellzhen) [inline src/main/resources/plugin.yml:38]: I think it should be /<command> [material] since it accepts a material as an arg

**Review by mitchellzhen — COMMENTED** [review]
Looks good overall, simpler and easier to read compared to before. Added 2 minor comments. Is this also compatible with 1.17?

File: src/main/java/com/multicraft/MultiCraftCommandExecutor.java (hunk 1 above)
Comment (mendozatudares) [reply src/main/java/com/multicraft/MultiCraftCommandExecutor.java:231]: That makes sense, I can make this into individual functions.

**Comment by mendozatudares** [comment]
> Looks good overall, simpler and easier to read compared to before. Added 2 minor comments. Is this also compatible with 1.17?

The 1.14 API is pretty much the same as the v1.17 API, I don't think there is any change necessary to the code to make it compatible with 1.17, but the Bukkit dependency would need to be bumped.

**Pull request merged**